### PR TITLE
React capstone project - Metrics-webapp

### PR DESCRIPTION
**Pull request by myelin0**
## This React capstone project is about building a mobile web application to check a list of metrics (numeric values) created by making use of React and Redux.

- Select an API that provides numeric data about a topic that you like and then build the webapp around it. The webapp will have several pages:  
>- The homepage is one page with a list of items that could be filtered by some parameters; 
>- One page for the item details.

- Technical requirements:

> -The project is a single-page application (SPA) built with React and Redux.
> -The data retrieved from the API should be stored in the Redux store.
> -You should filter the data that you retrieve from the API using a Filter stateless component.
> -Every page (the main page and the pages for each item) should have a unique route within the SPA.
> -The project should be deployed and accessible online (add a link to the online version of your app to your README file).

- Implement Testing with jest and React Testing Library.
- No linter errors.
- App explanation video:
https://www.loom.com/share/1e2c8af1537c4865b745dfab36d1d63b
(correction: There are 195 countries in the world today.)

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -1,10 +1,58 @@
...
+      <Query query={query} queryChangeHandler={queryHandler} />
+        <ul className="wrap">
+          {filteredData.map((country) => (
+            <li key={country.name} className="country-card p-2">
+              <Link to="/detail" className="name">
+                <h5
+                  className="name py-2"
+                  id={country.name}
+                  onClick={countryDetailsHandler}
+                >
```
Comment (EricMbouwe) [inline src/components/Home.js]: - [x] This function never gets the id parameter as an argument when it is called. To solve that you should use a callback function like ` onClick={() => countryDetailsHandler(country)}`

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @myelin0, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/Home.js (hunk 1 above)
Comment (myelin0) [reply src/components/Home.js]: Hello @EricMbouwe! Thank you for your review! 
I believe the `countryDetailsHandler` is clarified in lines 24-27 of the Home.js file.
 `const countryDetailsHandler = ({ target }) => {
    const { id } = target;
    dispatch(countryDetails(id));
  };`

**Review by Deepakdanger — APPROVED** [review]
Hi @myelin0, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**